src/utils/Scaffold.js

```diff
@@ -23,16 +25,23 @@ export default function Scaffold(sequence = [], _checkFn, { onDone, onChange } =
     let isDone = false
     let didStart = false
 
-    function onError(err) {
-        // eslint-disable-next-line require-atomic-updates
-        error = AggregatedError.from(error, err)
+    function collectErrors(err) {
+        try {
+            if (typeof onError === 'function') {
```

Comment:
Maybe it would be enough to check if `onError` is defined (but not to check if it is a function or not). 

Passing a non-function as `onError` parameter is a configuration error. If we need to be super-protective, we can validate the type in constructor. TypeScript will solve this anyway.

Reply:
Sure, perhaps makes sense to validate this upfront.

Reply:
Ah that's right the reason I don't do this is because then you need to decide on a system for doing runtime type validation errors, and it would want to be consistent. Easier to just do this.